devicetypes/smartthings/ikea-button.src/ikea-button.groovy

```diff
@@ -188,13 +192,17 @@ def installed() {
 		numberOfButtons = 3
 	}  else if (isSomfySituo4()) {
 		numberOfButtons = 12
+	} else if (isBSM300()) {
+		numberOfButtons = 1
+	} else if (isMSM300()) {
+		numberOfButtons = 4
 	}
 
 	if (numberOfButtons > 1) {
 		createChildButtonDevices(numberOfButtons)
 	}
 
-	def supportedButtons = isIkeaOpenCloseRemote() || isSomfy() ? ["pushed"] : ["pushed", "held"]
+	def supportedButtons = (isBSM300() || isMSM300()) ? ["pushed","held","double"] : (isIkeaOpenCloseRemote() || isSomfy() ? ["pushed"] : ["pushed", "held"])
```

Comment:
same here.

doubly-nested ternary operators are kind of hard to read

Reply:
@greens I have updated.